server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/BaseImageMapper.java

```diff
@@ -0,0 +1,164 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+/**
+ This class allows the mapping of OsCapabilities (from a tosca model) to a docker base image.
+ When initialized with Spring this class will automatically download the latest tags from the base images
+ defined in the <code>DockerBaseImages</code> every 24 hours (by default).
+ To update this value please modify the <code>toscana.docker.base-image-mapper.update-interval</code> property (value in
+ hours)
+ */
+@Service
+@Profile("!" + Profiles.EXCLUDE_BASE_IMAGE_MAPPER)
+@SuppressWarnings("ConstantConditions")
+public class BaseImageMapper {
+
+    private static final Logger logger = LoggerFactory.getLogger(BaseImageMapper.class);
+
+    /**
+     Stores the update intervall for the base Image mappings (in hours)
+     Taken from the property value:
+     <c
+     */
+    @Value("${toscana.docker.base-image-mapper.update-interval}")
+    private int updateInterval;
+
+    /**
+     This field is used to check the last update timestamp
+     */
+    private long lastUpdate;
+    /**
+     The map containing the "raw" tag data collected from dockerhub
+     */
+    private Map<String, DockerImage> imageMap = new HashMap<>();
+
+    /**
+     A array conatinig the base images that this Base Image Mapper tries to map to
+     this usually is <code>DockerBaseImages.values()</code>
+     */
+    private final DockerBaseImages[] baseImages;
+
+    private MapperEngine engine;
+
+    @Autowired
+    public BaseImageMapper(DockerBaseImages[] dockerBaseImages) {
+        this.baseImages = dockerBaseImages;
+        engine = new MapperEngine(imageMap);
+    }
+
+    /**
+     Toggles the first update of the mappings during initialisation in spring
+     */
+    @PostConstruct
+    private void postConstruct() {
+        updateBaseImageMap();
+    }
+
+    /**
+     Performs the Update of each image and sets the "last update" timestamp at the end
+     */
+    private void updateBaseImageMap() {
+        logger.info("Updating BaseImage Mappings");
+        for (DockerBaseImages baseImage : baseImages) {
+            logger.debug("Fetching tags for base image {}", baseImage.name());
+            List<ImageTags> tags = fetchImageTags(baseImage);
+            if (tags != null) {
+                addImagesForType(baseImage, tags);
+            }
+        }
+        lastUpdate = System.currentTimeMillis();
+        logger.info("Mappings have been updated. The next update will be executed in approx. {} hours", updateInterval);
+    }
+
+    /**
+     This is the spring scheduled job used to perform the updates after the frist one
+     <p>
```

Comment:
please close the paragraph

Reply:
if i recall correctly thats not needed when messing with javadocs